File: tasks/december/29.12/walk.js
function isValidWalk(walk) {
    const obj = {};
    for (const x of walk) {
        if (obj[x] === undefined) {
            obj[x] = 0;
        }
        obj[x] += 1;
    }
    const result = walk.length === 10 && obj.n === obj.s && obj.e === obj.w;
    return result;
}

module.exports = isValidWalk;

File: tasks/2023/december/29.12/walk.py
def is_valid_walk(walk):
    object = {
        'w': 0,
        'e': 0,
        's': 0,
        'n': 0
    }
    for i in walk:
        object[i] += 1
    return len(walk) == 10 and object['w'] == object['e'] and object['s'] == object['n']